wrappers/src/shared_realm_cs.cpp

```diff
@@ -128,7 +128,13 @@ Realm::Config get_shared_realm_config(Configuration configuration, SyncConfigura
 
 inline SharedRealm* new_realm(SharedRealm realm)
 {
-    if (!realm->refresh()) {
+    if (!realm->config().immutable()) {
+        if (!realm->refresh()) {
+            // this should only be needed for immutable frozen realms
```

Comment:
[nit] This comment is not correct or it's placed in the wrong line - if we're here, then the Realm we're operating on is not immutable.